### PR TITLE
Feat add tolerations and affinity to operator deployment

### PR DESCRIPTION
**Pull request by baptman21**
### Description

This PR adds the `affinity` and `tolerations` values to the Helm Chart for the deployment of the Operator.

### Usage Example

```yaml
operator:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
          - matchExpressions:
            - key: kubernetes.io/arch
              operator: In
              values:
              - amd64
  tolerations:
  - key: "vmtype"
    operator: "Equal"
    value: "spot"
    effect: "NoSchedule"
```

### Community Note

* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### CODE REVIEW

**Review by arybolovlev — COMMENTED** [review]
Hi @baptman21,

Thank you for your interest in contributing to this operator. I have left a few minor comments for you to address, which shouldn't take more than a minute.

Once you have made the changes, please update the Helm chart README by running `make helm-docs`. Then, generate a changelog entry with ```changie new --kind ENHANCEMENTS --custom "PR=495" --body '`Helm Chart`: Add the ability to configure `affinity` and `tolerations` for the Deployment of the operator.'```.

File: charts/hcp-terraform-operator/values.yaml [hunk 1]
```diff
@@ -44,6 +44,32 @@ operator:
...
+  # Usage example:
+  # tolerations:
+  # - key: "kubernetes.azure.com/scalesetpriority"
+  #   operator: "Equal"
+  #   value: "spot"
+  #   effect: "NoSchedule"
+  #
+  # -- Kubernetes Tolerations.
+  # More information: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+  tolerations: {}
```
[the remark this answers is not in the document]
Comment (baptman21) [reply charts/hcp-terraform-operator/values.yaml]: Done.

File: charts/hcp-terraform-operator/values.yaml [hunk 2]
```diff
@@ -44,6 +44,32 @@ operator:
...
+  #     requiredDuringSchedulingIgnoredDuringExecution:
+  #       nodeSelectorTerms:
+  #       - matchExpressions:
+  #         - key: kubernetes.io/arch
+  #           operator: In
+  #           values:
+  #         - amd64
+  #
+  # -- Kubernetes Affinity.
+  # More information: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
```
[the remark this answers is not in the document]
Comment (baptman21) [reply charts/hcp-terraform-operator/values.yaml]: Done.

File: charts/hcp-terraform-operator/values.yaml [hunk 3]
```diff
@@ -44,6 +44,32 @@ operator:
...
+
+  # Usage example:
+  # tolerations:
+  # - key: "kubernetes.azure.com/scalesetpriority"
+  #   operator: "Equal"
+  #   value: "spot"
+  #   effect: "NoSchedule"
+  #
+  # -- Kubernetes Tolerations.
+  # More information: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
```
[the remark this answers is not in the document]
Comment (baptman21) [reply charts/hcp-terraform-operator/values.yaml]: Done.

**Comment by arybolovlev** [comment]
Hi @baptman21,

Please let me know if you are willing to work on this PR and address the comments I have left.

Thank you.

**Comment by baptman21** [comment]
Hello @arybolovlev,

Good timing, I was just working on your comments. Made the necessary changes. Tell me if you need anything else.

Thanks,

**Review by arybolovlev — APPROVED** [review]
Thank you, @baptman21! 🚀

**Pull request merged**